### PR TITLE
Use medians in aggregate row instead of sum

### LINKED ISSUE #222: Change summary row to show medians, not sums
> Screenshot 2024-02-29 at 10 36 00 AM

_Originally posted by @jmelot in https://github.com/georgetown-cset/parat/issues/218#issuecomment-1971401376_

### PR DESCRIPTION
**Pull request by brianlove**
Instead of summing up all column values, calculate and display the median value.

Closes #222

### CODE REVIEW

**Review by jmelot — COMMENTED** [review]
Everything looks ok to me except that median is overapplied - we actually only need it in the columns I selected in #222 (plus "AI patents recent growth"). The reason we need it in these columns but not others is either:

a. they're related to publication counts - companies collaborate on papers in a way they don't on patents and jobs, so the double counting is only an issue on papers
b. they're percentages, where a median makes sense and a total does not 

One other thing - I think we should format the numbers in the total row with commas and percent signs as needed for consistency.

I'll fix the tests presently...

**Comment by brianlove** [comment]
Test fixes were split out and merged in #228.

Column definitions now have an additional option `isPercent` (columns with `isGrowthStat` are also treated as `isPercent === true`)

image

File: web/gui-v2/src/components/ListViewTable.jsx [hunk 1]
```diff
@@ -239,6 +257,16 @@ columnDefinitions.forEach((colDef) => {
     } else {
       SLIDER_NATURAL_COLUMNS.push(colDef.key);
     }
+
+    if ( colDef?.isGrowthStat || colDef?.isPercent ) {
+      SLIDER_PERCENTAGE_COLUMNS.push(colDef.key);
+    }
+
+    if ( colDef?.aggregateType === 'sum' ) {
```
Comment (jmelot) [inline web/gui-v2/src/components/ListViewTable.jsx:265]: I suggest we make this a default so we don't have to specify it in the majority of cases

**Review by jmelot — COMMENTED** [review]
This looks right, thanks! One fussy comment but I'm going to go ahead and merge

**Review by jmelot — APPROVED** [review]

**Pull request merged**